zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/ZkClient.java

```diff
@@ -1503,12 +1555,14 @@ private void acquireEventLock() {
         } catch (Exception e) {
           throw ExceptionUtil.convertToRuntimeException(e);
         }
+
         // before attempting a retry, check whether retry timeout has elapsed
         if (System.currentTimeMillis() - operationStartTime > _operationRetryTimeoutInMillis) {
           throw new ZkTimeoutException("Operation cannot be retried because of retry timeout ("
               + _operationRetryTimeoutInMillis + " milli seconds). Retry was caused by "
               + retryCauseCode);
         }
+        LOG.warn("Retrying operation, caused by {}", retryCauseCode);
```

Comment:
I think we removed this because many clients complaining that this retry log floods their log file. Please make it debug or don't add it.

Reply:
If you change it to debug, we shall be able to see it during the debug practice (by changing the log4j attribute). So Let's do debug for now.